joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1384,11 +1378,17 @@ bool JointTrajectoryController::validate_trajectory_msg(
       !validate_trajectory_point_field(joint_count, points[i].positions, "positions", i, false) ||
       !validate_trajectory_point_field(joint_count, points[i].velocities, "velocities", i, true) ||
       !validate_trajectory_point_field(
-        joint_count, points[i].accelerations, "accelerations", i, true) ||
-      !validate_trajectory_point_field(joint_count, points[i].effort, "effort", i, true))
+        joint_count, points[i].accelerations, "accelerations", i, true))
     {
       return false;
     }
+    // reject effort entries
+    if (points[i].effort.size() > 0)
```

Comment:
Optional nit, I feel it reads better this way

```suggestion
    if (!points[i].effort.empty())
```